modules/drivers/redshift/src/metabase/driver/redshift.clj

```diff
@@ -118,6 +118,7 @@
     (try
       (sql-jdbc.execute/set-best-transaction-level! driver conn)
       (sql-jdbc.execute/set-time-zone-if-supported! driver conn timezone-id)
+      (.setReadOnly conn true)
```

Comment:
the main reason we haven't done this in the past is because we reuse our connections in a connection pool, and I don't think Postgres (or Redshift for that matter) lets you `SET SESSION TIME ZONE` on a read-only connection. So this might work the first time around and then fail when that Connection gets re-used...

I'm not sure if that's actually the case or not, but it would be good to test this stuff out. Or maybe we can make the connect _not_ read-only before setting session timezone then switch it back to read-only... not sure if that's allowed or not

Reply:
As far as testing goes, do we have a way to ensure that the same connection is borrowed from the pool on a second query?  I suppose it should be possible to set the pool size to 1 just for the test run, which should force that behavior.

Reply:
Added a test in the updated commit that I believe gets at this.

Reply:
I thought the main reason for readOnly not being set was it blocked temporary tables from being created in a transaction.

Reply:
Yeah, it would definitely prevent temp tables from being created.  So it really seems like we can't blanket-enable this.  Maybe we can turn it off only for native queries?  (As far as I know, our regular MBQL flow shouldn't need to create them).  Or else make it a config option (not ideal UX, but maximum flexibility and security).  Something else?

Reply:
We're not creating temporary tables anywhere ourselves. I haven't heard anything about anyone else trying to do it, although I could be out of the loop. I guess that could be an issue in native queries. But the issue with setting session timezone is the main reason we didn't make stuff read-only in the past that I'm aware of

@jeff303 If you want to test `SET SESSION TIME ZONE` just get a Connection out of the pool (you can use `(jdbc/get-connection pool-spec)` to do this) and the make it read-only and see if you can change the timezone afterwards. That should lay this question to rest